CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+cmake_minimum_required(VERSION 3.0)
+project(aws-c-common)
+
+set(PROJECT_NAME aws-c-common)
+
+file(GLOB AWS_COMMON_HEADERS
+        "include/aws/common/*.h"
+        )
+
+file(GLOB AWS_COMMON_PRIV_HEADERS
+        "include/aws/common/private/*.h"
+        )
+
+file(GLOB AWS_COMMON_SRC
+        "source/*.c"
+        )
+
+if (WIN32)
+    file(GLOB AWS_COMMON_OS_SRC
+            "source/windows/*.c"
+            )
+
+    if (MSVC)
+        source_group("Header Files\\aws\\common" FILES ${AWS_COMMON_HEADERS})
+        source_group("Header Files\\aws\\common\\private" FILES ${AWS_COMMON_PRIV_HEADERS})
+        source_group("Source Files" FILES ${AWS_COMMON_SRC})
+        source_group("Source Files\\windows" FILES ${AWS_COMMON_OS_SRC})
+    endif ()
+
+    set(PLATFORM_LIBS "Kernel32")
+else ()
+    file(GLOB AWS_COMMON_OS_SRC
+            "source/posix/*.c"
+            )
+
+    set(PLATFORM_LIBS "pthread" "rt")
```

Comment:
librt doesn't exist on macos

Reply:
I used this cmake snippet to probe for these libraries when playing around with macos, if it helps.

```
CHECK_LIBRARY_EXISTS("rt" "clock_gettime" "" HAVE_LIBRT)
if(HAVE_LIBRT)
    set(PLATFORM_LIBS ${PLATFORM_LIBS} "rt")
endif()

CHECK_LIBRARY_EXISTS("pthread" "pthread_mutex_lock" "" HAVE_LIBPTHREAD)
if(HAVE_LIBRT)
    set(PLATFORM_LIBS ${PLATFORM_LIBS} "pthread")
endif()
```